### PR TITLE
Raspberry Pi GPIO door opener

### PR DESCRIPTION
**Pull request by ruippeixotog**
Opens a door by setting a GPIO pin to high during a set period of time.

### CODE REVIEW

File: door-open-service/node-backend/src/RpiGpioOpener.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+class RpiGpioOpener {
+
+  constructor(config) {
+    gpio.setMode(gpio.MODE_BCM);
+    this.pins = config.pins;
+    this.opening = {};
+  }
+
+  canOpen(doorId) {
+    return this.pins[doorId];
```
Comment (queimadus) [inline door-open-service/node-backend/src/RpiGpioOpener.js]: Return a boolean here instead with `!!this.pins[doorId]`. Is `0` a valid pin number? If so, this needs to handle this.

File: door-open-service/node-backend/src/RpiGpioOpener.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  }
+
+  open(doorId, time) {
```
Comment (queimadus) [inline door-open-service/node-backend/src/RpiGpioOpener.js]: As discussed, this method in the interface can later be improved to return the result.

File: door-open-service/node-backend/src/RpiGpioOpener.js (hunk 1 above)
Comment (ruippeixotog) [reply door-open-service/node-backend/src/RpiGpioOpener.js]: Nice catch, fixed in e19ee88.

**Comment by queimadus** [comment]
LGTM

**Pull request merged**